### PR TITLE
Fix paperless and other Python pkgs

### PR DESCRIPTION
**Pull request by erikarvstedt**
This fixes:
- django_extensions. The old version was incompatible with the newly updated version of `factory_boy`,
- django-picklefield: The source was missing test files.
- paperless: The pkg needed older versions of `factory_boy` and `django_extensions`.
  This also fixes the corresponding NixOS test.

This is the complete closure of dependents/referrers of all changed pkgs:

```
django_extensions -----------------\
                                    |---> hyperkitty
django-picklefield --> django-q  --/

paperless --> (none, leaf node)

```
All pkgs in this graph are now building again. This PR is strictly fixing pkgs and introduces no new breakage.

### CODE REVIEW

**Comment by seppeljordan** [comment]
Result of `nixpkgs-review pr 97283` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>11 packages built:</summary>
<br>- mailman-web (python38Packages.mailman-web)
<br>- paperless
<br>- python37Packages.django-picklefield
<br>- python37Packages.django-q
<br>- python37Packages.django_extensions
<br>- python37Packages.hyperkitty
<br>- python37Packages.mailman-web
<br>- python38Packages.django-picklefield
<br>- python38Packages.django-q
<br>- python38Packages.django_extensions
<br>- python38Packages.hyperkitty
</details>

File: pkgs/applications/office/paperless/python-modules/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+pyPkgs: fetchFromGitHub:
+{
+  django_2_0 = pyPkgs.django_2_2.overrideDerivation (_: rec {
```
Comment (jonringer) [inline pkgs/applications/office/paperless/python-modules/default.nix:3]: overrideDerivation should be avoided, please use `overrideAttrs`, or in the case of python packages, `overridePythonAttrs`

File: pkgs/development/python-modules/django-picklefield/default.nix [hunk 2]
```diff
@@ -11,6 +11,9 @@ buildPythonPackage rec {
 
   propagatedBuildInputs = [ django ];
 
+  # The PyPi source is missing test files
+  doCheck = false;
+
```
Comment (jonringer) [inline pkgs/development/python-modules/django-picklefield/default.nix:16]: There are no tests included. Please try to checkout from source and check if they have unit tests, and try to run them. Unit tests give a good indication that they package has a high degree of validity and correctness given the python package set.

If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors. Please see [pythonImportsCheck documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck) for more information.

File: pkgs/development/python-modules/django-picklefield/default.nix [hunk 3]
```diff
@@ -1,16 +1,27 @@
...
+    rev = "v${version}";
+    sha256 = "0ni7bc86k0ra4pc8zv451pzlpkhs1nyil1sq9jdb4m2mib87b5fk";
   };
 
   propagatedBuildInputs = [ django ];
 
+  checkInputs = [ pytest pytest-django ];
+
+  checkPhase = ''
+    PYTHONPATH="$(pwd):$PYTHONPATH" \
```
Comment (jonringer) [inline pkgs/development/python-modules/django-picklefield/default.nix:20]: this is default behavior of `python`, but doesn't really affect the build

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 97283 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>11 packages built:</summary>  <ul>    <li>mailman-web (python38Packages.mailman-web)</li>    <li>paperless</li>    <li>python37Packages.django-picklefield</li>    <li>python37Packages.django-q</li>    <li>python37Packages.django_extensions</li>    <li>python37Packages.hyperkitty</li>    <li>python37Packages.mailman-web</li>    <li>python38Packages.django-picklefield</li>    <li>python38Packages.django-q</li>    <li>python38Packages.django_extensions</li>    <li>python38Packages.hyperkitty</li>  </ul></details>

**Comment by jonringer** [comment]
please backport this :)

**Comment by jonringer** [comment]
ZHF: #97479

File: pkgs/development/python-modules/django-picklefield/default.nix (hunk 3 above)
Comment (erikarvstedt) [reply pkgs/development/python-modules/django-picklefield/default.nix:20]: No, when removing `PYTHONPATH="$(pwd):$PYTHONPATH"`, the test fails with `ImportError: No module named 'tests'`.

**Pull request merged**